meshsync/exec.go

```diff
@@ -43,18 +43,56 @@ func (h *Handler) processExecRequest(obj interface{}, cfg config.ListenerConfig)
 			if !bool(req.Stop) {
 				h.channelPool[id] = channels.NewStructChannel()
 				h.Log.Info("Starting session")
+
+				err := h.Broker.Publish("active_sessions.exec", &broker.Message{
+					ObjectType: broker.ActiveExecObject,
+					Object:     h.getActiveChannels(),
+				})
+				if err != nil {
+					h.Log.Error(ErrGetObject(err))
+				}
 				go h.streamSession(id, req, cfg)
 			}
 		} else {
 			// Already running subscription
 			if bool(req.Stop) {
+				// TODO: once we have a unsubscribe functionality, need to publish message to active sessions subject
 				execCleanup(h, id)
 			}
 		}
 	}
 
 	return nil
 }
+func (h *Handler) processActiveExecRequest() error {
+	go h.streamChannelPool()
+
+	return nil
+}
+func (h *Handler) getActiveChannels() []*string {
+	activeChannels := make([]*string, 0, len(h.channelPool))
+	for k := range h.channelPool {
+		activeChannels = append(activeChannels, &k)
+	}
+
+	return activeChannels
+}
+
+func (h *Handler) streamChannelPool() error {
+	go func() {
+		for {
```

Comment:
I think we can sleep in this loop for ~ 10sec or something. This would loop too quickly and would emit events too quickly in my opinion.

Reply:
@bariqhibat any progress here?